src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -115,29 +120,62 @@ private static IVsReferenceItem GetReferenceItem(KeyValuePair<String, IImmutable
             };
         }
 
-        private static IVsReferenceItems GetReferences(IImmutableDictionary<String, IImmutableDictionary<String, String>> items)
+        private static bool HasVersionAttribute(IImmutableDictionary<string, string> value)
         {
-            return new ReferenceItems(items.Select(p => GetReferenceItem(p)));
+            return value.TryGetValue(PackageReference.VersionProperty, out string version) 
+                && !string.IsNullOrEmpty(version);
         }
 
-        private static IVsReferenceItems GetProjectReferences(IImmutableDictionary<String, IImmutableDictionary<String, String>> items)
+        private static IVsReferenceItems GetPackageReferences(
+            IImmutableDictionary<string, IImmutableDictionary<string, string>> items)
+        {            
+            return new ReferenceItems(items
+                .Where(p => HasVersionAttribute(p.Value))
+                .Select(v => GetReferenceItem(v)));
+        }
+
+        private static IVsReferenceItems GetProjectReferences(
+            IImmutableDictionary<string, IImmutableDictionary<string, string>> projectReferenceItems,
+            IImmutableDictionary<string, IImmutableDictionary<string, string>> packageReferenceItems,
+            UnconfiguredProject project)
         {
-            var referenceItems = GetReferences(items);
-            foreach (ReferenceItem item in referenceItems)
+            var referenceItems = new ReferenceItems(projectReferenceItems.Select(p => GetReferenceItem(p)));
+
+            // include package references with no 'Version' attribute
+            var packageProjects = packageReferenceItems.Where(p => !HasVersionAttribute(p.Value));
+            foreach (var packageProjectItem in packageProjects)
             {
-                var definingProjectDirectory = item.Properties.Item(DefiningProjectDirectoryProperty);
-                string fullPathFromDefining = definingProjectDirectory != null
-                    ? Path.Combine(definingProjectDirectory.Value, item.Name)
-                    : item.Name;
+                referenceItems.Add(GetReferenceItem(packageProjectItem));
```

Comment:
can move `GetReferenceItem(packageProjectItem)` to the previous linq statement and `referenceItems.Add` can just add the resultant